crates/sovereign-sdk/module-system/sov-modules-api/src/default_signature.rs

```diff
@@ -333,6 +334,24 @@ impl BorshSerialize for DefaultPublicKey {
     }
 }
 
+impl StateKeyCodec<DefaultPublicKey> for BorshCodec {
+    fn encode_key(&self, value: &DefaultPublicKey) -> Vec<u8> {
+        borsh::to_vec(value).expect("Failed to serialize key")
+    }
+}
+
+impl StateValueCodec<DefaultPublicKey> for BorshCodec {
+    type Error = std::io::Error;
+
+    fn encode_value(&self, value: &DefaultPublicKey) -> Vec<u8> {
+        borsh::to_vec(&value).unwrap()
+    }
```

Comment:
can prealloc exact

Reply:
It's already preallocated because it's [u8;32] inside and borsh sees it

Reply:
If you go into borsh::to_vec fn though, it allocates 1024 sized Vec blindly to use as writer, what im suggesting is to create the writer Vec ourselves with exact size. I know its not that important of a optimization, but these things accumulate and becomes hard to detect since its scattered all over the place.